### PR TITLE
Adding some more route53 resources

### PR DESCRIPTION
**Pull request by zerbitx**
Internally we needed these resources nuked.  Hopefully they're of use to you as well.

### CODE REVIEW

File: resources/route53-resolver-endpoints.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+// Route53ResolverEndpoint is the resource type for nuking
+type Route53ResolverEndpoint struct {
+	svc  *route53resolver.Route53Resolver
+	id   *string
+	name *string
+	ips  []*route53resolver.IpAddressUpdate
+}
+
+func init() {
+	register("Route53ResolverEndpoints", ListRoute53ResolverEndpoints)
```
Comment (der-eismann) [inline resources/route53-resolver-endpoints.go:20]:
```suggestion
	register("Route53ResolverEndpoint", ListRoute53ResolverEndpoints)
```

We always use the singular for a resource.

File: resources/route53-resolver-endpoints.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+func ListRoute53ResolverEndpoints(sess *session.Session) ([]Resource, error) {
+	svc := route53resolver.New(sess)
+
+	params := &route53resolver.ListResolverEndpointsInput{}
+
+	resources := make([]Resource, 0)
+	output, err := svc.ListResolverEndpoints(params)
+
+	if err != nil {
+		return resources, err
```
Comment (der-eismann) [inline resources/route53-resolver-endpoints.go:33]:
```suggestion
		return nil, err
```

File: resources/route53-resolver-endpoints.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			name: endpoint.Name,
+		}
+
+		ipsOutput, err := svc.ListResolverEndpointIpAddresses(
+			&route53resolver.ListResolverEndpointIpAddressesInput{
+				ResolverEndpointId: endpoint.Id,
+			})
+
+		if err != nil {
+			return resources, err
```
Comment (der-eismann) [inline resources/route53-resolver-endpoints.go:49]:
```suggestion
			return nil, err
```

File: resources/route53-resolver-endpoints.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+				Ip:       ip.Ip,
+				IpId:     ip.IpId,
+				SubnetId: ip.SubnetId,
+			})
+		}
+
+		resources = append(resources, resolverEndpoint)
+	}
+
+	return resources, err
```
Comment (der-eismann) [inline resources/route53-resolver-endpoints.go:63]:
```suggestion
	return resources, nil
```

File: resources/route53-resolver-endpoints.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+			return resources, err
+		}
+
+		for _, ip := range ipsOutput.IpAddresses {
+			resolverEndpoint.ips = append(resolverEndpoint.ips, &route53resolver.IpAddressUpdate{
+				Ip:       ip.Ip,
+				IpId:     ip.IpId,
+				SubnetId: ip.SubnetId,
+			})
+		}
```
Comment (der-eismann) [inline resources/route53-resolver-endpoints.go:58]: What do you actually need these for? They are not used in the output or remove function.

File: resources/route53-resolver-rules.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	association struct {
+		id    *string
+		vpcID *string
+	}
+)
+
+func init() {
+	register("Route53ResolverRules", ListRout53ResolverRules)
```
Comment (der-eismann) [inline resources/route53-resolver-rules.go:28]:
```suggestion
	register("Route53ResolverRule", ListRoute53ResolverRules)
```

File: resources/route53-resolver-rules.go [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+// ListRout53ResolverRules produces the resources to be nuked.
+func ListRout53ResolverRules(sess *session.Session) ([]Resource, error) {
```
Comment (der-eismann) [inline resources/route53-resolver-rules.go:32]:
```suggestion
func ListRoute53ResolverRules(sess *session.Session) ([]Resource, error) {
```

File: resources/route53-resolver-rules.go [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+	svc := route53resolver.New(sess)
+
+	var resources []Resource
+	output, err := svc.ListResolverRules(&route53resolver.ListResolverRulesInput{})
+
+	if err != nil {
+		return resources, err
```
Comment (der-eismann) [inline resources/route53-resolver-rules.go:39]:
```suggestion
		return nil, err
```

File: resources/route53-resolver-rules.go [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	associationsOutput, err := svc.ListResolverRuleAssociations(&route53resolver.ListResolverRuleAssociationsInput{})
+
+	if err != nil {
+		return resources, err
```
Comment (der-eismann) [inline resources/route53-resolver-rules.go:45]:
```suggestion
		return nil, err
```

File: resources/route53-resolver-rules.go [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+		id         *string
+		name       *string
+		domainName *string
+		vpcIds     []*string
+	}
+
+	association struct {
+		id    *string
+		vpcID *string
+	}
```
Comment (der-eismann) [inline resources/route53-resolver-rules.go:24]: AFAICS this struct isn't used anywhere, is it?

File: resources/route53-resolver-rules.go [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+		ResolverRuleId: r.id,
+	})
+
+	return err
+}
+
+// Properties provides debugging output
+func (r *Route53ResolverRule) Properties() types.Properties {
+	return types.NewProperties().
+		Set("Id", r.id).
```
Comment (der-eismann) [inline resources/route53-resolver-rules.go:107]:
```suggestion
		Set("ID", r.id).
```

File: resources/route53-traffic-policies.go [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+	return instances, nil
+}
+
+type Route53TrafficPolicy struct {
+	svc       *route53.Route53
+	id        *string
+	name      *string
+	version   *int64
+	instances []*route53.TrafficPolicyInstance
+}
```
Comment (der-eismann) [inline resources/route53-traffic-policies.go:86]: Please put this at the top like in the other resources.

File: resources/route53-traffic-policies.go [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+	_, err := tp.svc.DeleteTrafficPolicy(params)
+
+	return err
+}
+
+func (tp *Route53TrafficPolicy) Properties() types.Properties {
+	return types.NewProperties().
+		Set("ID", *tp.id).
+		Set("NAME", *tp.name)
```
Comment (der-eismann) [inline resources/route53-traffic-policies.go:112]:
```suggestion
		Set("Name", *tp.name)
```

**Review by der-eismann — CHANGES_REQUESTED** [review]

**Comment by der-eismann** [comment]
Closes #508

File: resources/route53-resolver-endpoints.go (hunk 5 above)
Comment (zerbitx) [reply resources/route53-resolver-endpoints.go:58]: Ah, left in from initial dev when we did have them in the output for a time.  I'll take them out, thanks!

File: resources/route53-resolver-rules.go (hunk 10 above)
Comment (zerbitx) [reply resources/route53-resolver-rules.go:24]: Nope, good catch.

File: resources/route53-resolver-rules.go (hunk 7 above)
Comment (zerbitx) [reply resources/route53-resolver-rules.go]: Good eye!

File: resources/route53-traffic-policies.go [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+	_, err := tp.svc.DeleteTrafficPolicy(params)
+
+	return err
+}
+
+func (tp *Route53TrafficPolicy) Properties() types.Properties {
+	return types.NewProperties().
+		Set("ID", *tp.id).
+		Set("Name", *tp.name)
+}
```
Comment (der-eismann) [inline resources/route53-traffic-policies.go:113]: Sorry, completely missed this the first time. Could you please add a `String()` function for traffic policies?

**Review by der-eismann — DISMISSED** [review]
Everything else is looking good now. Thanks for the changes!

**Review by svenwltr — APPROVED** [review]

**Review by bjoernhaeuser — APPROVED** [review]

**Pull request merged**